core/src/main/scala/cats/FlatMap.scala

```diff
@@ -137,6 +137,22 @@ trait FlatMap[F[_]] extends Apply[F] with FlatMapArityFunctions[F] {
 
   /**
    * `if` lifted into monad.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.Eval
+   * scala> import cats.syntax.all._
+   *
+   * scala> val b1 = Eval.now(true)
+   * scala> val asBool1 = b1.ifM(Eval.now(true), Eval.now(false))
```

Comment:
I think this example would be more clear if the items on the right were not also booleans. What about 
```scala
b1.ifM(Eval.now("it's true!"), Eval.now("it's false. :("))
```
or something like that.

If they are all booleans it can confuse someone into thinking that may be required.

Reply:
That's a good point, I updated it to use Strings.